ui/ducks/bridge/actions.ts

```diff
@@ -70,3 +104,356 @@ export const setToChain = (chainId: Hex) => {
     );
   };
 };
+
+export const getBridgeERC20Allowance = async (
+  contractAddress: string,
+  walletAddress: string,
+  chainId: Hex,
+): ReturnType<BridgeController['getBridgeERC20Allowance']> => {
+  return await submitRequestToBackground(
+    BridgeBackgroundAction.GET_BRIDGE_ERC20_ALLOWANCE,
+    [contractAddress, walletAddress, chainId],
+  );
+};
+
+export const submitBridgeTransaction = (
+  history: ReturnType<typeof useHistory>,
+) => {
+  return async (
+    dispatch: MetaMaskReduxDispatch,
+    getState: () => MetaMaskReduxState & BridgeAppState,
+  ) => {
+    const state = getState();
+    const isBridgeEnabled = getIsBridgeEnabled(state);
+    const isBridgeChain = getIsBridgeChain(state);
+
+    if (!(isBridgeEnabled && isBridgeChain)) {
+      // TODO do we want to do something here?
+      return;
+    }
+
+    const quoteMetas = getQuotes(state);
+    const quoteMeta = quoteMetas[0];
+
+    // Track event TODO
+
+    const calcFeePerGas = () => {
+      let maxFeePerGas: undefined | string;
+      let maxPriorityFeePerGas: undefined | string;
+
+      const networkAndAccountSupports1559 =
+        checkNetworkAndAccountSupports1559(state);
+
+      if (networkAndAccountSupports1559) {
+        const gasFeeEstimates = getGasFeeEstimates(state);
+        maxFeePerGas = decGWEIToHexWEI(
+          gasFeeEstimates?.high?.suggestedMaxFeePerGas,
+        );
+        maxPriorityFeePerGas = decGWEIToHexWEI(
+          gasFeeEstimates?.high?.suggestedMaxPriorityFeePerGas,
+        );
+      }
+
+      return {
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+      };
+    };
+
+    const calcMaxGasLimit = (gasLimit: number, gasMultiplier = 1) => {
+      return new Numeric(
+        new BigNumber(gasLimit).times(gasMultiplier).round(0).toString(),
+        10,
+      ).toPrefixedHexString();
+    };
```

Comment:
Do the transactions still succeed without applying gas multipliers? The bridge-api applies them as well so the provided gasLimits are pre-adjusted

Reply:
Still seems to work, I can remove this then

Reply:
Done in [e9116bb](https://github.com/MetaMask/metamask-extension/pull/27262/commits/e9116bbc27cf9208491725b5b38cab6700aec615)